ubjw.c

```diff
@@ -231,6 +227,11 @@ static inline void priv_ubjw_tag_public(ubjw_context_t* ctx, UBJ_TYPE tid)
 		{
 			ch->elements_remaining--; //todo: error if elements remaining is 0;
 		}
+
+		if ((ch->flags & CONTAINER_IS_TYPED) && ch->type == tid)
+		{
+			return;
+		}

```

Comment:
This needs to be moved down so that `CONTAINER_EXPECTS_KEY` is turned back for objects and the size is updated correctly in typed containers.